### PR TITLE
bpo-29636: json.tool: Add document for indentation options.

### PR DESCRIPTION
**Pull request by dhimmel**
https://github.com/python/cpython/pull/345 added indentation options for the `json.tool` command line utility. This PR follows up with some additional modifications that I didn't have time to put in the previous PR before it was merged.

https://bugs.python.org/issue29636

CC @methane

### CODE REVIEW

File: Doc/library/json.rst [hunk 1]
```diff
@@ -738,6 +738,12 @@ Command line options
 
    .. versionadded:: 3.8
 
+.. cmdoption:: --indent, --tab, --no-indent, --compact
+
+   Mutually exclusive options for whitespace control
+
+   .. versionadded:: 3.9
+
```
Comment (dhimmel) [inline Doc/library/json.rst:752]: @wimglenn does this documentation look okay? I combined multiple arguments into a single line (not sure if that violates convention or is okay)

File: Lib/json/tool.py [hunk 2]
```diff
@@ -31,7 +31,8 @@ def main():
     parser.add_argument('--sort-keys', action='store_true', default=False,
                         help='sort the output of dictionaries alphabetically by key')
     parser.add_argument('--json-lines', action='store_true', default=False,
-                        help='parse input using the jsonlines format')
+                        help='parse input using the jsonlines format. '
+                        'Use with --no-indent or --compact to produce valid jsonlines output.')
```
Comment (dhimmel) [inline Lib/json/tool.py]: @hongweipeng (author of https://github.com/python/cpython/pull/10051): do you think this is a worthwhile and accurate addition to the documentation now that indentation options are supported?

Comment (hongweipeng) [reply Lib/json/tool.py]: Looks good. Thanks.

**Review by methane — APPROVED** [review]

**Comment by dhimmel** [comment]
Rendered documentation available at https://docs.python.org/3.9/library/json.html#cmdoption-json-tool-indent

> image

I forgot a period after "whitespace control" ):

**Pull request merged**